src/QsCompiler/Transformations/CallGraph/CallGraphBase.cs

```diff
@@ -0,0 +1,255 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.Quantum.QsCompiler.SyntaxTree;
+
+#nullable enable
+
+namespace Microsoft.Quantum.QsCompiler.DependencyAnalysis
+{
+    /// <summary>
+    /// Base class for call graph edge types.
+    /// </summary>
+    public abstract class CallGraphEdgeBase : IEquatable<CallGraphEdgeBase>
+    {
+        /// <summary>
+        /// Name of the callable where the reference was made.
+        /// </summary>
+        public QsQualifiedName FromCallableName { get; }
+
+        /// <summary>
+        /// Name of the callable being referenced.
+        /// </summary>
+        public QsQualifiedName ToCallableName { get; }
+
+        /// <summary>
+        /// The range of the reference represented by the edge.
+        /// </summary>
+        public DataTypes.Range ReferenceRange { get; }
+
+        /// <summary>
+        /// Base constructor for call graph edges. Initializes BaseCallGraphEdge properties.
+        /// Throws an ArgumentNullException if any of the arguments are null.
+        /// </summary>
+        protected CallGraphEdgeBase(QsQualifiedName fromCallableName, QsQualifiedName toCallableName, DataTypes.Range referenceRange)
+        {
+            if (fromCallableName is null)
+            {
+                throw new ArgumentNullException(nameof(fromCallableName));
+            }
+
+            if (toCallableName is null)
+            {
+                throw new ArgumentNullException(nameof(toCallableName));
+            }
+
+            if (referenceRange is null)
+            {
+                throw new ArgumentNullException(nameof(referenceRange));
+            }
+
+            this.FromCallableName = fromCallableName;
+            this.ToCallableName = toCallableName;
+            this.ReferenceRange = referenceRange;
+        }
+
+        /// <summary>
+        /// Determines if the object is the same as the given edge, ignoring the
+        /// ordering of key-value pairs in the type parameter dictionaries.
```

Comment:
This comment is referring to a type parameter dictionary that the base edges don't have.

Reply:
Removed